spectral_cube/cube_utils.py

```diff
@@ -230,3 +232,42 @@ def iterator_strategy(cube, axis=None):
     if cube.size < 1e8:  # smallish
         return 'cube'
     return 'slice'
+
+
+def beams_to_bintable(beams):
+    """
+    Convert a list of beams to a CASA-style BinTableHDU
+    """
+    metakeys = beams[0].meta.keys()
+    dtypes = ([('BMAJ','float'),
+               ('BMIN','float'),
+               ('BPA','float'),] +
+              [(key, type(val)) for key,val in beams[0].meta.items()]
+             )
+    beam_table = np.recarray(len(beams), dtype=dtypes)
+    beam_table['BMAJ'] = [bm.major.to(u.arcsec).value for bm in beams]
+    beam_table['BMIN'] = [bm.minor.to(u.arcsec).value for bm in beams]
+    beam_table['BPA'] = [bm.pa.to(u.deg).value for bm in beams]
+    for key in metakeys:
+        beam_table[key] = [bm.meta[key] for bm in beams]
+
+    bmhdu = BinTableHDU(beam_table)
+    return bmhdu
+
+def average_beams(beams):
+    """
+    Average the beam major, minor, and PA attributes.
+
+    This is usually a dumb thing to do!
+    """
+
+    from radio_beam import Beam
+    from astropy.stats import circmean
+
+    major = u.Quantity([bm.major for bm in beams], u.deg)
+    minor = u.Quantity([bm.minor for bm in beams], u.deg)
+    pa = u.Quantity([bm.pa for bm in beams], u.deg)
+    new_beam = Beam(major=major.mean(), minor=minor.mean(),
+                    pa=circmean(pa, weights=major/minor))
```

Comment:
That was my thinking.  I don't know what a correct weighting scheme would be; probably needs to be determined